vm-setup/roles/v1aX_integration_test/tasks/cleanup.yml

```diff
@@ -23,52 +23,70 @@
           (IMAGE_OS == "")
 
   - name: Check if cluster deprovisioning started.
-    shell: "kubectl get cluster -n {{ NAMESPACE }} -o json | jq -r '.items[] | .status.phase'"
+    k8s_info:
+      api_version: cluster.x-k8s.io/v1alpha3
+      kind: Cluster
+      namespace: "{{ NAMESPACE }}"
     register: deprovision_cluster
     retries: 2
     delay: 20
-    until: (deprovision_cluster.stdout == "deleting") or
-           (deprovision_cluster.stdout == "Deleting") or
-           (deprovision_cluster.stdout == "")
+    until: (deprovision_cluster is succeeded) and (
+             (deprovision_cluster.resources | length == 0) or
+             (deprovision_cluster.resources[0].status.phase | lower == "deleting"))
 
   - name: Wait until "{{ NUM_NODES }}" bmhs become ready again.
-    shell: |
-        kubectl get bmh -n {{ NAMESPACE }} -o json | jq -r '[ .items[]
-        | select (.status.provisioning.state == "ready")
-        | .metadata.name ] | length'
+    k8s_info:
+      api_version: metal3.io/v1alpha1
+      kind: BareMetalHost
+      namespace: "{{ NAMESPACE }}"
     register: deprovisioned_nodes
     retries: 150
     delay: 10
-    until: deprovisioned_nodes.stdout ==  NUM_NODES
+    vars:
+      query: "[? status.provisioning.state=='ready'].status.provisioning.state"
+    until: (deprovisioned_nodes is succeeded) and
+           (deprovisioned_nodes.resources | json_query(query) | length == (NUM_NODES | int))
 
   - name: Wait until no metal3machines are remaining
-    shell: |
-        kubectl get metal3machine -n {{ NAMESPACE }} -o json | jq -r '[ .items[] ] | length'
+    k8s_info:
+      api_version: infrastructure.cluster.x-k8s.io/v1alpha4
```

Comment:
I think we also need to parameterize this since we would have two versions of these soon

Reply:
Done